Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -215,15 +215,30 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
 
       string GetDescription(Boogie.ProofObligationDescription? description) {
         switch (assertionNode?.StatusVerification) {
-          case GutterVerificationStatus.Verified:
-            return $"{obsolescence}<span style='color:green'>**Success:**</span> " +
-                   (description?.SuccessDescription ?? "_no message_");
+          case GutterVerificationStatus.Verified: {
+              var successDescription = description?.SuccessDescription ?? "_no message_";
+              if (successDescription == "error is impossible: This is the precondition that might not hold.") {
+                successDescription = "the precondition always holds";
```

Comment:
If we have the concept of success descriptions, shouldn't they be defined together with the failure descriptions? This seems difficult to maintain.